File: Python/Programmers/Leve1/행렬의 덧셈.py
# def solution(arr1, arr2):
#     answer = []
#     for i in range(len(arr1)):
#         tmp = []
#         for a, b in zip(arr1[i], arr2[i]): # zip을 이용하여 묶고 합을 계산
#             tmp.append(a+b)
        
#         answer.append(tmp)
    
#     return answer

# def solution(arr1, arr2):
#     answer = []
#     for i in range(len(arr1)):
#         temp = []
#         for j in range(len(arr1[i])):
#             temp.append(arr1[i][j] + arr2[i][j]) # 하니씩 더함
#         answer.append(temp)
    
#     return answer

def solution(arr1, arr2):
    for i in range(len(arr1)):
        for j in range(len(arr1[i])):
            arr1[i][j] += arr2[i][j] # arr1이나 arr2에 행렬 덧셈 값을 바로 저장
            
    return arr1

File: JavaScript/Programmers/Level 1/행렬의 덧셈.js
function solution(arr1, arr2) {
    // map 메소드를 이용하여 덧셈하여 mapping, 2차원 배열이므로 map 메소드를 2번 사용해야 각 요소에 접근할 수 있음
    // map 메소드를 이용하여 행을 가져오고, 각 행을 다시 map 메소드를 이용하여 각 요소를 가져옴, 이후 arr1, arr2의 각 요소를 더한 결과를 return
    return arr1.map((_, i) => arr1[i].map((_, j) => arr1[i][j] + arr2[i][j]));
}
